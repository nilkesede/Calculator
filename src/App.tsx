import React from 'react'
import {StatusBar} from 'react-native'
import styled, {ThemeProvider} from 'styled-components/native'

import {theme} from './config'
import {Display, Keyboard} from './components'

const View = styled.View`
  flex: 1;
`

export default function App() {
  return (
    <ThemeProvider theme={theme}>
      <StatusBar barStyle="light-content" />
      <View>
        <Display value={'123.456'} />
        <Keyboard />
      </View>
    </ThemeProvider>
  )
}
