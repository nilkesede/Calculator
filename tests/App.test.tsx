import React from 'react'
import {render} from './utils'

import App from '../src/App'

it('renders correctly', () => {
  const tree = render(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
