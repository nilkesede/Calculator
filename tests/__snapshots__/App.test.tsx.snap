// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#303030",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 100,
            "fontWeight": "100",
            "textAlign": "right",
          },
        ]
      }
    >
      123.456
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#303030",
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          AC
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          +/-
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          %
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          /
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          7
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          8
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          9
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          x
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          4
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          5
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          6
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          1
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          2
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          3
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        big={true}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 2,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          .
        </Text>
      </View>
      <View
        big={false}
        style={
          Array [
            Object {
              "backgroundColor": "#707070",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 1,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 1,
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
                "fontWeight": "200",
                "textAlign": "center",
              },
            ]
          }
        >
          =
        </Text>
      </View>
    </View>
  </View>
</View>
`;
